core-plugins/docs/Joiner-batchjoiner.md

```diff
@@ -1,49 +1,91 @@
 Join key from each input stage will be prefixed with `<stageName>.` and the relation among join keys from different inputs is represented by `=`. 
 For example: customers.customer_id=items.c_id&customers.customer_name=items.c_name means the join key is a composite key
 of customer id and customer name from customers and items input stages and join will be performed on equality 
 of the join keys. This transform only supports equality for joins.
 
-**selectedFields:** Comma-separated list of fields to be selected and renamed in join output from each input stage. 
-Each selected field that should be present in the output must be prefixed with '<input_stage_name>'. 
-The syntax for specifying alias for each selected field is similar to sql. 
-For example: customers.id as customer_id, customer.name as customer_name, item.id as item_id, <stageName>.inputFieldName as alias. 
-The output will have same order of fields as selected in selectedFields. There must not be any duplicate fields in output.
-
-**requiredInputs:** Comma-separated list of stages. Required input stages decide the type of the join. 
-If all the input stages are present in required inputs, inner join will be performed. 
-Otherwise, outer join will be performed considering non-required inputs as optional.
-
-**numPartitions:** Number of partitions to use when grouping fields. If not specified, the execution
+**Inputs to Load in Memory:** Hint to the underlying execution engine that the specified input data should be
+loaded into memory to perform an in-memory join. This is ignored by the MapReduce engine and passed onto the Spark engine.
+An in-memory join performs well when one side of the join is small (for example, under 1gb). Be sure to set
+Spark executor memory to a number large enough to load all of these datasets into memory. This is most commonly
+used when a large input is being joined to a small input and will lead to much better performance in such scenarios.
+
+**Join on Null Keys:** Whether to join rows together if both of their key values are null.
+For example, suppose the join is on a 'purchases' input that contains:
+
+| purchase_id | customer_name | item   |
+| ----------- | ------------- | ------ |
+| 1           | alice         | donut  |
+| 2           |               | coffee | 
+| 3           | bob           | water  |
+
+and a 'customers' input that contains:
+
+| customer_id | name   |
+| ----------- | ------ |
+| 1           | alice  |
+| 2           |        |
+| 3           | bob    |
+
+The join is a left outer join on purchases.customer_name = customers.name.
+If this property is set to true, the joined output would be:
+
+| purchase_id | customer_name | item   | customer_id | name  |
+| ----------- | ------------- | ------ | ----------- | ----- |
+| 1           | alice         | donut  | 1           | alice |
+| 2           |               | coffee | 2           |       |
+| 2           |               | coffee | 3           |       |
```

Comment:
Should this be 2 and next line be 3?

Reply:
yeah, thanks for catching, this row should be deleted. I changed the example input data and didn't change this.